src/sim.rs

```diff
@@ -547,6 +558,20 @@ mod test {
         Ok(())
     }
 
+    /// This is a regression test that ensures JoinError::Cancelled is not
+    /// getting propagated to the test when the host crashes.
+    /// Before the fix the error returned by run function causing test to fail.
```

Comment:
edit: ...is not propagated to the test when the host crashes, which was causing incorrect test failure.